### PR TITLE
Feature/46 cli poc

### PR DESCRIPTION
**Pull request by michalbabiarz**
cli-module ready for review and release after passing the review.
Test results from Unit and Integration: 
http://de-muc-devonfw-allure01:8888/job/test/job/mrchecker-cli-module/job/feature%252F46-cli-poc/10/

### CODE REVIEW

File: mrchecker-framework-modules/mrchecker-cli-module/src/resources/enviroments/environments.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+service variable,DEV,QA,PROD,DEV1,DEV2
+DB_USERNAME,tester,#qa_user#,,,
+DB_PASSWORD,Vx4hylOJhNzRc5kKm4gr,#qa_pass#,,,
+DB_CONNECTION_URL,localhost:3307/int_tests,#qa_host#:3306/int_tests,,,
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-cli-module/src/resources/enviroments/environments.csv]: Copy-paste issue. To be removed.

File: mrchecker-framework-modules/mrchecker-cli-module/src/resources/settings.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+selenium.phantomjs=./lib/webdrivers/phantomjs/bin/phantomjs.exe
+selenium.chrome=./lib/webdrivers/chrome/chromedriver.exe
+selenium.firefox=./lib/webdrivers/firefox/geckodriver.exe
+selenium.ie=./lib/webdrivers/internetexplorer/IEDriverServer.exe
```
Comment (michalbabiarz) [inline mrchecker-framework-modules/mrchecker-cli-module/src/resources/settings.properties]: Copy-paste issue. To be removed.

File: mrchecker-framework-modules/mrchecker-cli-module/pom.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        <name>Capgemini - Nearshore Test Center - Poland</name>
+        <url>http://nsc.capgemini.com/nearshoretestcenter</url>
+    </organization>
+
+    <developers>
+        <developer>
+            <name>Michal Babiarz</name>
+            <email>michal.babiarz@capgemini.com</email>
+            <organization>Capgemini - Nearshore Test Center - Poland</organization>
+            <organizationUrl>http://nsc.capgemini.com/nearshoretestcenter</organizationUrl>
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-cli-module/pom.xml:37]: This page doesn't exist anymore

File: mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+import com.capgemini.mrchecker.test.core.ModuleType;
+import com.capgemini.mrchecker.test.core.Page;
+import com.capgemini.mrchecker.test.core.base.environment.IEnvironmentService;
+import com.capgemini.mrchecker.test.core.logger.BFLogger;
+import com.capgemini.mrchecker.test.core.logger.BFLoggerInstance;
+
+abstract public class BasePageCli extends Page {
+	
+	private static IEnvironmentService environmentService;
+
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java:26]: Is java code formatter on?

File: mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+	}
+	
+	@Override
+	public void onTestClassFinish() {
+		super.onTestClassFinish();
+		closeStreams();
+		closeProcess();
+		while((Objects.nonNull(responseWatcher) && responseWatcher.isAlive()) || (Objects.nonNull(errorWatcher) && errorWatcher.isAlive())) {
+			try {
+				Thread.sleep(100);
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java]: Is this sleep OK?

File: mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+	
+	public boolean isAlive() {
+		return (Objects.nonNull(process) && process.isAlive()) || (Objects.nonNull(responseWatcher) && responseWatcher.isAlive()) || (Objects.nonNull(errorWatcher) && errorWatcher.isAlive());
+	}
+	
+	public void waitToFinish() {
+		try {
+			process.waitFor();
+			while (isAlive()) {
+				Thread.sleep(100);
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java]: Is this sleep OK?

File: mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+			this.reader = reader;
+			this.output = output;
+		}
+		
+		@Override
+		public void run() {
+			try {
+				while (Objects.nonNull(process) && (process.isAlive() || reader.ready())) {
+					output.append(getLinesFromReader(reader));
+					Thread.sleep(100);
```
Comment (MalgorzataDzienia) [inline mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java:251]: Is this sleep OK?

**Review by MalgorzataDzienia — APPROVED** [review]
@Pyskaczu  Please check my review questions.

File: mrchecker-framework-modules/mrchecker-cli-module/src/resources/enviroments/environments.csv (hunk 1 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-cli-module/src/resources/enviroments/environments.csv]: removed

File: mrchecker-framework-modules/mrchecker-cli-module/src/resources/settings.properties (hunk 2 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-cli-module/src/resources/settings.properties]: removed

File: mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java (hunk 4 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-cli-module/src/main/java/com/capgemini/mrchecker/cli/core/BasePageCli.java:26]: Right, it was not.

File: mrchecker-framework-modules/mrchecker-cli-module/pom.xml (hunk 3 above)
Comment (michalbabiarz) [reply mrchecker-framework-modules/mrchecker-cli-module/pom.xml:37]: @MalgorzataDzienia  This is present in all the modules. What do we want to do with that? Remove or replace with some other URL?

Comment (MalgorzataDzienia) [reply mrchecker-framework-modules/mrchecker-cli-module/pom.xml:37]: https://capgeminisoftware.pl/

The name is also not right. We are Software Solution Center

**Pull request merged**